tests/test_vrf.py

```diff
@@ -31,6 +31,11 @@
     So, we prefer a fixture rather than xunit-style setup/teardown functions.
 """
 
+@pytest.fixture(scope="module", autouse=True)
+def setup_check_topo(testbed):
+    if testbed['topo']['name'] != 't0':
```

Comment:
why use topo name here, not type?

Reply:
Because we meet setup error on t0-56 as below

```
>       g_vars['vrf_intfs'] = get_vrf_intfs(cfg_facts)

test_vrf.py:426:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

cfg_facts = {'ACL_TABLE': {'DATAACL': {'policy_desc': 'DATAACL', 'ports': ['PortChannel0001', 'PortChannel0002', 'PortChannel0003'...BUFFER_POOL': {'ingress_lossy_pool': {'mode': 'dynamic', 'size': '32736769', 'type': 'ingress', 'xoff': '19457'}}, ...}

    def get_vrf_intfs(cfg_facts):
        intf_tables = ['INTERFACE', 'PORTCHANNEL_INTERFACE', 'VLAN_INTERFACE', 'LOOPBACK_INTERFACE']
        vrf_intfs = {}

        for table in intf_tables:
            for intf, attrs in cfg_facts.get(table, {}).iteritems():
                if '|' not in intf:
>                   vrf = attrs['vrf_name']
E                   KeyError: 'vrf_name'

test_vrf.py:115: KeyError
```

Reply:
the test should be fixed to support t0-56. you can open an issue to ask support vrf support in t0-56. but we should have uniform rule to decide whether a test can or cannot be run on certain topology. otherwise, the maintenance cost is too high.